pkg/apis/core/validation/shoot.go

```diff
@@ -681,12 +681,18 @@ func validateKubeControllerManager(kubernetesVersion string, kcm *core.KubeContr
 	if err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, kubernetesVersion, err.Error()))
 	}
+
 	if kcm != nil {
 		if maskSize := kcm.NodeCIDRMaskSize; maskSize != nil {
 			if *maskSize < 16 || *maskSize > 28 {
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("nodeCIDRMaskSize"), *maskSize, "nodeCIDRMaskSize must be between 16 and 28"))
 			}
 		}
+
+		if podEvictionTimeout := kcm.PodEvictionTimeout; podEvictionTimeout != nil && podEvictionTimeout.Duration == 0 {
```

Comment:
also validate for negative durations - I could set:
```
kubeControllerManager:
 nodeCIDRMaskSize: 24
 podEvictionTimeout: -3m0s
```

Reply:
Curiously the kube-controller-manager pod could startup and was running fine. So I am not sure

Reply:
It makes sense to validate that it's non negative and > 0, so I've added this condition now and an additional test for it. Thanks, please check.